lib/postProcess.js

```diff
@@ -47,9 +47,14 @@ postProcess._fetchRelatedResources = function(request, mainResource, callback) {
 
   resourcesToFetch = Object.keys(resourcesToFetch).map(function(type) {
     var ids = resourcesToFetch[type];
-    var urlJoiner = "&filter[id]=";
-    ids = urlJoiner + ids.join(urlJoiner);
-    return jsonApi._apiConfig.pathPrefix + type + "/?" + ids;
+    var qsField = "filter[id]=";
+    var urlJoiner = "&" + qsField;
+    ids = qsField + ids.join(urlJoiner);
+    var uri = jsonApi._apiConfig.pathPrefix + type + "/?" + ids;
+    if (request.route.query) {
+      uri += "&" + request.route.query;
+    }
+    return uri;
```

Comment:
Is this change designed to try and remove the leading `&` from the query string? It's quite hard to read by comparison, I'd rather we either live with the leading ampersand or we simply substring it off with a comment saying `// trim the leading ampersand`.